soaktest/soaktest.go

```diff
@@ -0,0 +1,240 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"time"
+)
+
+var DrErrCnt uint64
+var DrCnt uint64
+var producerErrorCbCnt uint64
+var MsgDupCntForOutput uint64
+var MsgMissCntForOutput uint64
+var ProducerMsgId uint64
+var ConsumerMsgCnt uint64
+var ConsumerMsgErrs uint64
+var ConsumerErr uint64
+var ProducerErr uint64
+
+var PRODUCER_DR_TO = "producer.dr."
+var PRODUCER_DR_ERR_TO = "producer.dr,err."
+
+const CONSUMER_ERROR = "consumer.err."
+const PRODUCER_ERROR = "producer.err."
+const FAILED_TO_PRODUCE_MESSAGE_TO = "failed.to.produce.message.to."
+const CONSUMER_RECEIVE_FROM = "consumer.receive.from."
+const CONSUMER_RECEIVE_ERROR_FROM = "consumer.receive.err.from."
+const PRODUCER_DROK = "producer.drok"
+const PRODUCER_LATENCY = "producer.latency"
+const LATENCY = "latency."
+const MSG_DUP_FROM = "msg.dup.from."
+const MSG_MISS_FROM = "msg.miss.from."
+const BROKER_RTT_P99 = "broker.rtt.p99."
+const BROKER_RTT_AVG = "broker.rtt.avg."
+
+// rttStats extracts broker rtt statistics from the raw map,
+// monitors broker avg and p99 using datadog
+func rttStats(raw map[string]interface{}, topic string) bool {
+	for _, broker := range raw["brokers"].(map[string]interface{}) {
+		var parts string
+		var avg float64
+		var p99 float64
+		var nodeid int
+
+		topparsValue := broker.(map[string]interface{})["toppars"]
+		if topparsValue == nil ||
+			len(topparsValue.(map[string]interface{})) == 0 {
+			continue
+		}
+
+		for _, val := range topparsValue.(map[string]interface{}) {
+			if val.(map[string]interface{})["topic"] != topic {
+				continue
+			}
+			partition := val.(map[string]interface{})["partition"]
+			f, ok := partition.(float64)
+			if !ok {
+				fmt.Printf("Failed to convert partition"+
+					"to float64 %v\n", ok)
+				return ok
+			}
+			if len(parts) > 0 {
+				parts += ","
+			}
+			parts += fmt.Sprintf("%d", int(f))
+		}
+
+		rttValue := broker.(map[string]interface{})["rtt"]
+		avg, ok := rttValue.(map[string]interface{})["avg"].(float64)
+		if !ok {
+			fmt.Printf("Failed to convert avg to float64 %v\n", ok)
+			return ok
+		}
+
+		p99, ok = rttValue.(map[string]interface{})["p99"].(float64)
+		if !ok {
+			fmt.Printf("Failed to convert p99 to float64 %v\n", ok)
+			return ok
+		}
+
+		nid, ok := broker.(map[string]interface{})["nodeid"].(float64)
+		if !ok {
+			fmt.Printf("Failed to convert nodeid to float64 %v\n", ok)
+			return ok
+		}
+		nodeid = int(nid)
+
+		tags := []string{fmt.Sprintf("broker:%d, type: %s",
```

Comment:
I think these should be two different tags perhaps?
Tags are usually static names, e.g. "broker", "type", "partition", while the tag value is the variable part.